### PR TITLE
refactor UCT transports

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8236/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8239/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8241/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8242/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8243/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8250/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8251/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8252/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@Artemy-Mellanox let's split this to several phases.
First PR - just remove dc_verbs transport and update gtestm without any refactoring.
Next PRs - refactoring DC, each PR up to 500 lines

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8258/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8297/ for details (Mellanox internal link).

File: src/uct/ib/rc/accel/rc_mlx5_struct.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2016.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCT_RC_MLX5_STRUCT_H
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_struct.h]: why need to move the structs from rc_mlx5_common.h to this file?

File: src/ucp/core/ucp_context.c [hunk 2]
```diff
@@ -242,12 +241,12 @@ UCS_CONFIG_REGISTER_TABLE(ucp_config_table, "UCP context", NULL, ucp_config_t)
...
   { "sm",    { "mm", "knem", "cma", "rdmacm", NULL } },
   { "shm",   { "mm", "knem", "cma", "rdmacm", NULL } },
-  { "ib",    { "rc", "ud", "dc", "rc_mlx5", "ud_mlx5", "dc_mlx5", "rdmacm", NULL } },
+  { "ib",    { "rc", "ud", "rc_mlx5", "ud_mlx5", "dc_mlx5", "rdmacm", NULL } },
   { "ud",    { "ud", "rdmacm", NULL } },
   { "ud_x",  { "ud_mlx5", "rdmacm", NULL } },
   { "rc",    { "rc", "ud:aux", "rdmacm", NULL } },
   { "rc_x",  { "rc_mlx5", "ud_mlx5:aux", "rdmacm", NULL } },
-  { "dc",    { "dc", "rdmacm", NULL } },
+  { "dc",    { "dc_mlx5", "rdmacm", NULL } },
```
Comment (yosefe) [inline src/ucp/core/ucp_context.c]: 👍

File: src/uct/ib/rc/accel/rc_mlx5_struct.h (hunk 1 above)
Comment (Artemy-Mellanox) [reply src/uct/ib/rc/accel/rc_mlx5_struct.h]: test_obj_size.cc includes DC headers. It can't include entire rc_mlx5_common.h.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8302/ for details (Mellanox internal link).

**Comment by Artemy-Mellanox** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8305/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8306/ for details (Mellanox internal link).

**Pull request merged**